assets/css/_route_picker.scss

```diff
@@ -1,92 +1,135 @@
 .m-route-picker {
+  flex-grow: 1;
   display: flex;
   flex-direction: column;
-  height: 100%;
+  row-gap: 1rem;
+  min-height: 0;
 }
 
 .m-route-picker__label {
   @include font-label();
 }
 
-.m-route-picker__selected-routes {
-  display: flex;
-  flex: 0 0 auto;
-  flex-shrink: 0;
-  flex-wrap: wrap;
-  margin-bottom: 1.5rem;
-  margin-top: 1.5rem;
-
-  &:empty {
-    margin-bottom: 0rem;
+.m-route-picker__routes-container {
+  flex-grow: 1;
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  grid-auto-rows: 1fr;
+  grid-template-areas: "all selected";
+  gap: 1rem;
+  min-height: 0;
+
+  ul {
+    min-height: 0;
+    background-color: $white;
+    border: 1px solid #b3b5bd;
+    border-top-left-radius: 6px;
+    border-bottom-left-radius: 6px;
+    margin: 0;
+    padding: 0.5rem;
+    overflow-y: scroll;
+    display: flex;
+    flex-direction: column;
+    gap: 0.5rem;
+
+    li {
+      position: relative;
+    }
+  }
+
+  p {
+    @include font-small;
+    background-color: $white;
+    border: 1px solid #b3b5bd;
+    border-top-left-radius: 6px;
+    border-bottom-left-radius: 6px;
+    margin: 0;
+    padding: 0.5rem 0.59375rem 0 0.84375rem;
+    color: #525560;
+    line-height: calc(4 / 3);
   }
 }
 
-.m-route-picker__selected-routes-button {
-  @include button-primary;
-  margin-bottom: 0.25rem;
-  margin-right: 0.25rem;
-  min-width: 2rem;
+@mixin route-picker-button {
+  width: 100%;
+  font-size: 0.875rem;
+  font-weight: 500;
+  height: 2.25rem;
+  border-radius: 4px;
+  line-height: 1;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  &:hover,
+  &:focus {
+    font-weight: 600;
+  }
 }
 
 .m-route-picker__route-list {
-  flex: 1 1 0;
-  overflow-y: scroll;
-  margin-top: 0rem;
+  grid-area: all;
 }
 
 .m-route-picker__route-list-button {
-  margin-bottom: 0.25rem;
-  width: 100%;
-  word-break: break-word;
+  @include route-picker-button;
+  color: #3c3f4c;
+  background-color: #ecedee;
+  border: 1px solid #dbdde1;
 
-  &--selected {
-    @include button-primary;
+  &:hover {
+    color: #141524;
```

Comment:
**suggestion (non-blocking):** This value (and the line below) are used in a few places. It might be worthwhile turning them into variables.